flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormat.java

```diff
@@ -91,6 +101,19 @@ private void establishConnection() throws SQLException, ClassNotFoundException {
 		} else {
 			dbConn = DriverManager.getConnection(dbURL, username, password);
 		}
+		timer = new Timer();
+		timer.schedule(new TimerTask() {
+			@Override
+			public void run() {
+				try {
```

Comment:
Should we throw a runtime exceptions here? It might be a better idea to wait until the actual JDBC write to throw in case it is just transient, and might got recovered afterwards during batch write

Reply:
Thanks for the review. I agree with you that we shouldn't throw a exception here. Just update the code.